enhancements/agent-installer/automated-workflow-kubeconfig.md

```diff
@@ -0,0 +1,253 @@
+---
+title: automated-workflow-kubeconfig
+authors:
+  - "@zaneb"
+reviewers:
+  - "@patrickdillon"
+  - "@sadasu"
+  - "@avishayt"
+  - "@mhrivnak"
+  - "@andfasano"
+  - "@bfournie"
+  - "@lranjbar"
+  - "@pawanpinjarkar"
+  - "@rwsu"
+approvers:
+  - "@dhellmann"
+api-approvers:
+  - None
+creation-date: 2022-05-18
+last-updated: 2022-05-18
+tracking-link:
+  - https://issues.redhat.com/browse/AGENT-40
+see-also:
+  - "/enhancements/agent-installer/agent-based-installer.md"
+  - "/enhancements/agent-installer/automated-workflow-for-agent-based-installer.md"
+replaces: N/A
+superseded-by: N/A
+---
+
+# Pre-generating the admin Kubeconfig in the Agent-based installer
+
+## Summary
+
+We will generate the admin kubeconfig file in the agent installer at the same
+time as the agent ISO. The CA keys used to sign the client certificate in the
+kubeconfig will be added to the ISO and eventually installed into the cluster
+so that the credentials in the admin kubeconfig will be accepted.
+
+## Motivation
+
+In the hosted assisted service, the admin kubeconfig is available for download
+from the web UI only after assisted-service has run openshift-installer to
+create the bootstrap ignition (i.e. after cluster installation has started).
+Since the hosted service is always available, this config can be downloaded at
+any time before the certificates in it expire.
+
+For the agent-based installer, doing the same would mean the kubeconfig was
+only available for a brief window of time, since the assisted-service is
+ephemeral. It runs on node 0 of the cluster and disappears after bootstrapping
+when that node is rebooted to become a cluster member.
+
+Furthermore, the kubeconfig file needs to be available to the installer in
+order for it to be used to monitor the cluster status in the `openshift-install
+agent wait-for install-complete` command.
+
+### User Stories
+
+* As a cluster creator, I want to have a reliable, non-timing dependent way to
+  retrieve the admin kubeconfig for a new cluster created with the agent-based
+  installer.
+
+* As a cluster creator, I want to use the `agent wait-for install-complete`
+  command to wait for the cluster to complete, with the preceding steps in the
+  automated workflow limited to only running the `agent create image` command
+  and booting the ISO on the hosts.
+
+### Goals
+
+* Generate a working admin kubeconfig as part of the `openshift-install agent
+  create image` command.
+
+### Non-Goals
```

Comment:
Should making the kubeadmin password available be a non-goal?

Reply:
Why wouldn't we include that password, too?

Reply:
As someone who never uses the console, TBH I'm not sure how this works in a _normal_ installation. It appears that in IPI/UPI it is written to the file `auth/kubeadmin-password`? What, if anything, does assisted do? Presumably it's easy enough to set a new password if you have the admin kubeconfig.

Reply:
It is available via REST API and UI just like the kubeconfig, and when running as an operator it is stored in a secret.

Reply:
hmm how exposing a kubeadmin password can be not a goal? We want to provide CUs friendly experience, rather than adding more steps for them to perform the installation. 
And yes, I assume most of the customers will go to console.